fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaData.java

```diff
@@ -396,6 +418,8 @@ private static void getDependencies(@Nonnull Descriptors.FileDescriptor fileDesc
         builder.setSplitLongRecords(splitLongRecords);
         builder.setStoreRecordVersions(storeRecordVersions);
         builder.setVersion(version);
+        builder.setSubspaceKeysCounter(subspaceKeysCounter);
+        builder.setUsesSubspaceKeysCounter(isCounterBasedSubspaceKeysEnabled());
```

Comment:
I left this comment on a previous version of this code:

> I believe in that if `!isIndexCounterBasedSubspaceKeyEnabled()` and the check to make sure that the `index_subspace_key_counter` is not set if the counter-based strategy is not enabled (referenced in this comment: [b117cf8#r257863558](https://github.com/FoundationDB/fdb-record-layer/commit/b117cf83ea6e81a4f7f7667988d4d1b6eca87634#r257863558)), then I think the meta-data will fail validation checks upon deserialization.

I think this is still true, or at least, the intent of the check I referenced seems to clash with the way these two lines are written.

Reply:
Yeah, I think this is adjusted to handle the case correctly now.